virttest/qemu_vm.py

```diff
@@ -1895,6 +1895,13 @@ def sort_key(dev):
         # Add usb devices
         for usb_dev in params.objects("usb_devices"):
             usb_dev_params = params.object_params(usb_dev)
+            if "usb-redir" == usb_dev_params.get("usb_type"):
+                # Add usb-redir device chardev backend
+                chardev_id = usb_dev_params.get("chardev_id",
+                                                "%s-chardev" % usb_dev)
+                char_device = devices.chardev_define_by_params(chardev_id,
+                                                               usb_dev_params)
+                devices.insert(char_device)
```

Comment:
Suggest move L1898~L1904 to qcontainer, it is what `usb_by_params` should do.
For `usb-redir`, add one usb device need two part, chardev and usb-redir

Reply:
Updated, thanks!